taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -520,15 +517,15 @@ func (self *Contract) DistributeRewards(blockAuthorAddr *common.Address, rewards
 
 	votesReward := big.NewInt(0)
 	blockAuthorReward := big.NewInt(0)
-	trxsReward := blockReward
+	dagProposersReward := blockReward
 	// We need to handle case for block 1
 	if rewardsStats.TotalVotesWeight > 0 {
 		// Calculate propotion between votes and transactions
-		trxsReward = bigutil.Div(bigutil.Mul(blockReward, TransactionsRewardPercentage), big.NewInt(100))
-		votesReward = bigutil.Sub(blockReward, trxsReward)
+		dagProposersReward = bigutil.Div(bigutil.Mul(blockReward, big.NewInt(int64(self.cfg.DagProposersReward))), big.NewInt(100))
+		votesReward = bigutil.Sub(blockReward, dagProposersReward)
 
-		// Calculate bonus reward based on MaxBlockAuthorReward and subtract from total votes reward
-		bonusReward := bigutil.Div(bigutil.Mul(votesReward, big.NewInt(int64(self.cfg.MaxBlockAuthorReward))), big.NewInt(100))
+		// Calculate bonus reward as part of blockReward multiplied by MaxBlockAuthorReward and subtract it from total votes reward part. As the reward part of Dag defined above and it should not change
+		bonusReward := bigutil.Div(bigutil.Mul(blockReward, big.NewInt(int64(self.cfg.MaxBlockAuthorReward))), big.NewInt(100))
```

Comment:
so we reserve 5% from blockReward as bonusReward.  Then if pbft block author does not include all of the possible votes, he should not get full bonusReward, see else branch here:
i```
f maxVotesWeigh == rewardsStats.TotalVotesWeight {
	blockAuthorReward = bonusReward
} else {
	twoTPlusOne := maxVotesWeigh*2/3 + 1
	bonusVotesWeight := rewardsStats.TotalVotesWeight - twoTPlusOne
	// should be zero if rewardsStats.TotalVotesWeight == twoTPlusOne
	blockAuthorReward = bigutil.Div(bigutil.Mul(bonusReward, big.NewInt(int64(bonusVotesWeight))), big.NewInt(int64(maxVotesWeigh-twoTPlusOne)))

```}

But we dont do anything else with this bonusReward afterwards (if block proposer does not get full bonus, the rest of the bonus should go somwhere but it doesnt), which means some part of bonus reward was just lost, doesnt it ?  

This check should then fail I guess ? 
```
if totalRewardCheck.Cmp(blockReward) == 1 {
  errorString := fmt.Sprintf("totalRewardCheck (%d) is more then blockReward (%d)", totalRewardCheck, blockReward)
  panic(errorString)
}
```

but it does not, how is this supposed to work ?

Reply:
It is working the same way it was before this changes. If we have not all bonus votes here, then part of `blockAuthorReward` would be not issued(burned). So `totalRewardCheck` would be smaller than originally calculated `blockReward`, so it is ok

Reply:
in that case something is not working because we never substract anything from blockReward. That means if we burn some rewards (do not reward anyone with it) this check should never pass ?
```
if totalRewardCheck.Cmp(blockReward) == 1 {
  errorString := fmt.Sprintf("totalRewardCheck (%d) is more then blockReward (%d)", totalRewardCheck, blockReward)
  panic(errorString)
}
```

Reply:
ah this check does not require it to be the same, totalRewardCheck must be < blockReward